### ISSUE TITLE
Issues with type mismatch and shell package

### ISSUE THREAD

**Issue by infinisil** [Infinisil/nix-xdg#2]
Separate issue for the ones mentioned by @orivej in #1:

> While testing this I encountered two errors. First, `xdgSpec = specs: specs.base;` did not work due to the wrong type:
> ```
> error: The option value `xdgSpec' in `/etc/nixos/configuration.nix' is not of type `attribute set of unspecifieds'.
> ```
> Second, I have `users.extraUsers.user.shell = pkgs.zsh` and the switch fails with
> ```
> error: /nix/store/xlzwmn1wy4l2x0437gcijxh5s07v35pw-zsh-5.4.2-xdg is not a shell package
> ```

**Comment by infinisil** [comment]
Regarding the first one, I made the mistake of using `types.attrsOf` even though it can be a function, so this needs to be changed to `types.unspecified`.

Regarding the second one, apparently this happens due to passthru attributes not getting passed through.

I fixed these in https://github.com/Infinisil/nix-xdg/commit/230fb8ae96f6660e2c2990accdbbf1764942ff11, although I haven't tested the shell thing, I'm pretty sure it should work though.
